test/e2e/shared/aws_helpers.go

```diff
@@ -0,0 +1,176 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shared
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/client"
+	rgapi "github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/api/v1beta1"
+)
+
+type LoadBalancerType string
+
+var (
+	LoadBalancerTypeELB = LoadBalancerType("elb")
+	LoadBalancerTypeALB = LoadBalancerType("alb")
+	LoadBalancerTypeNLB = LoadBalancerType("nlb")
+)
+
+type WaitForLoadBalancerToExistForServiceInput struct {
+	AWSSession       client.ConfigProvider
+	ServiceName      string
+	ServiceNamespace string
+	ClusterName      string
+	Type             LoadBalancerType
+}
+
+func WaitForLoadBalancerToExistForService(ctx context.Context, input WaitForLoadBalancerToExistForServiceInput, intervals ...interface{}) {
+	By(fmt.Sprintf("Waiting for AWS load balancer of type %s to exist for service %s/%s", input.Type, input.ServiceNamespace, input.ServiceName))
+
+	Eventually(func() bool {
+		arns, err := GetLoadBalancerARNs(ctx, GetLoadBalancerARNsInput{ //nolint: gosimple
+			AWSSession:       input.AWSSession,
+			ServiceName:      input.ServiceName,
+			ServiceNamespace: input.ServiceNamespace,
+			ClusterName:      input.ClusterName,
+			Type:             input.Type,
+		})
+		if err != nil {
+			fmt.Fprintf(GinkgoWriter, "error getting loadbalancer arns: %v\n", err)
+
+			return false
+		}
+		if len(arns) == 0 {
+			return false
+		}
+
+		return true
+	}, intervals...).Should(BeTrue(), "failed to wait for loadbalancer")
+}
+
+type GetLoadBalancerARNsInput struct {
+	AWSSession       client.ConfigProvider
+	ServiceName      string
+	ServiceNamespace string
+	ClusterName      string
+	Type             LoadBalancerType
+}
+
+func GetLoadBalancerARNs(ctx context.Context, input GetLoadBalancerARNsInput) ([]string, error) {
```

Comment:
I don't see context being used and this doesn't appear to implement any interfaces... soo... remove the context? :)

Reply:
Yep or use it.

Reply:
Removed it.